src/main/java/com/unstoppabledomains/resolution/naming/service/ENS.java

```diff
@@ -75,6 +75,35 @@ public String getOwner(String domain) throws NamingServiceException {
     return owner;
   }
 
+  @Override
+  public String[] batchOwners(String[] domains) throws NamingServiceException {
+    if (domains.length > 255) {
+      throw new NamingServiceException(NSExceptionCode.MaxThreadLimit, new NSExceptionParams("m|l", "Ens#batchOwners", "200"));
+    }
+    String[] owners = new String[domains.length];
+    Thread[] threads = new Thread[domains.length];
```

Comment:
You can use `List<Thread>` here and then simply call `add(new Thread(...))`

Reply:
Same with `owners`

Reply:
I need to maintain the correct order. With List instead of String[] one thread can finish later and will be added later to the owners' list than supposed to.
In this way, I have full control of what place in the Array I want to fill when a certain Thread has finished. 

I have updated the code to return List<String> but inside still have to use the String[], unless there is a way to specifically set the order of items in a list.

Reply:
@hivtushok Is there a way to achieve the specific order with Lists instead of String[] in case of multiple threads ?